src/xhr.js

```diff
@@ -348,6 +382,15 @@ class FetchResponse {
   json() {
     return this.drainText_().then(JSON.parse);
   }
+
+  /**
+   * Reads the xhr response. Could possibly be undefined if server
+   * does not return anything.
+   * @return {?HTMLDocument|undefined}
+   */
+  doc() {
```

Comment:
Since this method is not in spec, let's make it private.

You also need this here:
```
    dev.assert(!this.bodyUsed, 'Body already used');
    this.bodyUsed = true;
```